pkg/controller/servicebindingrequest/mapper.go

```diff
@@ -12,7 +21,97 @@ var (
 
 // sbrRequestMapper is the handler.Mapper interface implementation. It should influence the
 // enqueue process considering the resources informed.
-type sbrRequestMapper struct{}
+type sbrRequestMapper struct {
+	client     dynamic.Interface
+	restMapper meta.RESTMapper
+}
+
+var serviceBindingRequestGVK = v1alpha1.SchemeGroupVersion.WithKind("ServiceBindingRequest")
+var secretGVK = corev1.SchemeGroupVersion.WithKind("Secret")
+
+// isServiceBindingRequest checks whether the given obj is a Service Binding Request through GVK
+// comparison.
+func isServiceBindingRequest(obj runtime.Object) bool {
+	return obj.GetObjectKind().GroupVersionKind() == serviceBindingRequestGVK
+}
+
+// isSecret checks whether the given obj is a Secret through GVK comparison.
+func isSecret(obj runtime.Object) bool {
+	return obj.GetObjectKind().GroupVersionKind() == secretGVK
+}
+
+// isSBRService checks whether the given obj is a service in given sbr.
+func isSBRService(sbr *v1alpha1.ServiceBindingRequest, obj runtime.Object) bool {
+	services := extractServiceSelectors(sbr)
+	for _, svc := range services {
+		svcGVK := schema.GroupVersionKind{Group: svc.Group, Version: svc.Version, Kind: svc.Kind}
+		if obj.GetObjectKind().GroupVersionKind() == svcGVK {
+			return true
+		}
+	}
+	return false
+}
+
+// isApplication checks whether the given obj is an application in given sbr.
+func isApplication(
```

Comment:
We are using terminology like is**SBRService** which is nice. Then we should rename `isApplication` to is**SBRApplication** so that we know this is not just any application. This is a check for SBR's application.

Reply:
Solved in 34d695f6f7fa42e218ba58a7421436c77f175ae3.